DirectProgramming/DPC++FPGA/PlatformTests/board_test/src/board_test.hpp

```diff
@@ -0,0 +1,1852 @@
+            closedir(reports_dir);
+            return 1;
+          }  // End of if - else reading the json file
+
+          // No need to iterate through rest of directory once required file is
+          // found
+          break;
+
+        }  // End of if looking for quartus.json
+
+      }  // End of while reading file names inside reports/lib/json directory
+
+      if (!file_found) {
+        // Could not find quartus.json, report error and terminate test
+        std::cout
+            << "Cannot find " << file_name << " JSON file in the " << paths[0]
+            << " or " << paths[1] << "directory\n"
+            << "Failed to report Quartus compiled frequency, please see NOTE "
+            << "at the beginning of this kernel clock frequency test.\n"
+            << "Reporting measured frequency and terminating test.\n\n"
+            << "Measured Frequency = " << kernel_freq << "\n";
+        closedir(reports_dir);
+        return 1;
+      }  // End of if reporting file not found
+
+      // Successfully read Quartus compiled frequency, close reports directory
+      closedir(reports_dir);
+
+    }  // End of if - else to read data from reports directory
+  } else {
+    // USer has selected to not to compare measured frequency with Quartus
+    // compiled frequency, report and continue other tests
+    std::cout
+        << "*** NOTE ***: User has selected to turn off the comparison of "
+        << "measured frequency with Quartus compiled frequency by setting the "
+        << "\"report_chk\" variable to \"false\" in <board_test.cpp>\n"
+        << "The Quartus compiled frequency will not be reported and the "
+        << "remaining tests will run without this check\n"
+        << "Please see Note at the beginning of this kernel clock frequency "
+        << "test for more information about this setting.\n"
+        << "Reporting measured frequency and continuing remaining tests.\n"
+        << "Measured Frequency = " << kernel_freq << "\n";
+  }  // End of if - else to check for reports
+
+  return 0;
+}
+
+//////////////////////////////////////
+// **** KernelLatency function **** //
+//////////////////////////////////////
+
+// Inputs:
+// queue &q - queue to submit operation
+// Returns:
+// 0 indicating successful completion (no error checks)
+
+// The function does the following tasks:
+// 1. Launch large number of no operation kernels
+// 2. Measure total time for the above kernels to launch and finish
+// 3. Calculate kernel latency and kernel throughput based on total time and
+// number of kernels
+// 4. Report the latency and throughput and return
+
+int ShimMetrics::KernelLatency(queue &q) {
+  // **** Launch no-op kernel multiple times **** //
+  auto start = std::chrono::system_clock::now();
+  constexpr size_t kNumKernels = 10000;
+  for (size_t l = 0; l < kNumKernels; l++) {
+    auto e = q.submit([&](handler &h) {
+      h.single_task<NopSingleTask>([=]() {});  // no operation kernel
+    });
```

Comment:
`auto e = q.single_task<NopSingleTask>([=] {});`

You can use this other places as well.